src/core/libraries/kernel/file_system.cpp

```diff
@@ -359,6 +391,29 @@ s32 PS4_SYSV_ABI sceKernelFsync(int fd) {
     return ORBIS_OK;
 }
 
+int PS4_SYSV_ABI sceKernelFtruncate(int fd, s64 length) {
+    auto* h = Common::Singleton<Core::FileSys::HandleTable>::Instance();
+    auto* file = h->GetFile(fd);
+
+    if (file == nullptr) {
+        return SCE_KERNEL_ERROR_EBADF;
+    }
+
+    if (file->m_host_name.empty()) {
+        return SCE_KERNEL_ERROR_EACCES;
+    }
+
+    // CUSA05289 (PSPHD emulator): Tries to resize a temporary file after deleting it
+    try {
+        // TODO: Check for sufficient space?
+        std::filesystem::resize_file(file->m_host_name, length);
```

Comment:
It would be better to use an overload which returns `std::error_code` instead of exception handling